### PR TITLE
[Runtime] #ifdef __clang__ on printf __attribute__ in Portability.h

### PR DESCRIPTION
**Pull request by mikeash**
I broke the Windows build again.

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]
We might be able to make this a bit looser and include GCC, but lets get the build repaired first (and there are other known issues with GCC right now).

File: include/swift/Runtime/Portability.h [hunk 1]
```diff
@@ -28,7 +28,10 @@ size_t _swift_strlcpy(char *dst, const char *src, size_t maxlen);
 #ifdef SWIFT_RUNTIME_ATTRIBUTE_ALWAYS_INLINE
 SWIFT_RUNTIME_ATTRIBUTE_ALWAYS_INLINE
 #endif
-__attribute((__format__(__printf__, 2, 3))) inline static int
+#ifdef __clang__
```
Comment (gwynne) [inline include/swift/Runtime/Portability.h:31]: Per `Config.h`, couldn't you just check `__has_attribute(format)`?

Comment (compnerd) [reply include/swift/Runtime/Portability.h:31]: Yeah, we could, but at that point, its probably better to introduce a bigger wrapper like the `SWIFT_RUNTIME_ATTRIBUTE_ALWAYS_INLINE`.

**Comment by davezarzycki** [comment]
This attribute is a very old GCCism and after playing around with godblot.org, it appears that only MSVC rejects this attribute. Every other compiler seems to be able to handle it just fine, warning/error free.

**Pull request merged**